internal/handlers/commonparams/extract_params.go

```diff
@@ -211,8 +268,87 @@ func setStructField(elem *reflect.Value, i int, command, key string, val any) er
 		if err != nil {
 			return err
 		}
-	case reflect.String, reflect.Bool, reflect.Struct, reflect.Pointer:
+	case reflect.String, reflect.Struct, reflect.Pointer, reflect.Interface:
+		settable = val
+	case reflect.Bool:
+		if o.numericBool {
+			settable, err = GetBoolOptionalParam(val, key)
+			if err != nil {
+				return err
+			}
+
+			break
+		}
+
+		if o.numericAsBool {
+			var numeric int64
+
+			numeric, err = GetWholeNumberParam(val)
+			if err != nil || numeric < 0 || numeric > 1 {
+				return commonerrors.NewCommandErrorMsgWithArgument(
+					commonerrors.ErrFailedToParse,
+					fmt.Sprintf("The '%s.%s' field must be 0 or 1. Got %v", command, key, types.FormatAnyValue(val)),
+					command,
+				)
+			}
+
+			settable = numeric == 1
+
+			break
+		}
+
 		settable = val
+	case reflect.Slice:
+		array, ok := val.(*types.Array)
+		if !ok {
+			return commonerrors.NewCommandErrorMsgWithArgument(
+				commonerrors.ErrTypeMismatch,
+				fmt.Sprintf(
+					`BSON field '%s.%s' is the wrong type '%s', expected type '%s'`,
+					command, key, AliasFromType(val), AliasFromType(fv.Interface()),
+				),
+				command,
+			)
+		}
+
+		iter := array.Iterator()
+		defer iter.Close()
+
+		arrayToSet := reflect.MakeSlice(fv.Type(), array.Len(), array.Len())
+
+		for {
+			i, arrayDoc, err := iter.Next()
+			if errors.Is(err, iterator.ErrIteratorDone) {
+				break
+			}
+
+			if err != nil {
+				return lazyerrors.Error(err)
+			}
+
+			doc, ok := arrayDoc.(*types.Document)
+			if !ok {
+				return commonerrors.NewCommandErrorMsgWithArgument(
+					commonerrors.ErrTypeMismatch,
+					fmt.Sprintf(
+						`BSON field '%s.%s' is the wrong type '%s', expected type '%s'`,
+						command, key, AliasFromType(val), AliasFromType(fv.Interface()),
+					),
+					command,
+				)
+			}
```

Comment:
In the future we may find this logic may be too strict. Array may contain elements that isn't document. But we don't have example parameter for now so it's probably ok as it is for now.

Reply:
Make it handled inside `checkAllRequiredFIeldsPopulated`.